ldap/servers/slapd/back-ldbm/db-bdb/bdb_layer.h

```diff
@@ -91,8 +91,9 @@ typedef struct bdb_config
     int bdb_previous_lock_config;  /* Max lock count when we last shut down--
                                       * used to determine if we delete the mpool */
     u_int32_t bdb_deadlock_policy; /* i.e. the atype to DB_ENV->lock_detect in bdb_deadlock_threadmain */
-    int bdb_compactdb_interval;    /* interval to execute compact id2entry dbs */
-    char *bdb_compactdb_time;       /* time of day to execute compact id2entry dbs */
+    int bdb_compactdb_interval;       /* interval to execute compact id2entry dbs */
+    char *bdb_compactdb_time;         /* time of day to execute compact id2entry dbs */
+    int32_t bdb_compactdb_starttime; /* the time the interval was started */
```

Comment:
Should we have it as int64_t right away to avoid 2038 issue? Or, probably, the correct option will be a `time_t`, unless I miss something.